util/labels/helpers.go

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package labels
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha4"
+)
+
+// HasWatchLabel returns true if the object has a label with the WatchLabel key and a configured value.
+func HasWatchLabel(o metav1.Object, labelValue string) bool {
+	labels := o.GetLabels()
+	if labels == nil {
+		return false
+	}
+	val, ok := labels[clusterv1.WatchLabel]
```

Comment:
It's safe to read from a `nil` map so this can be simplified a bit more.
```suggestion
	val, ok := o.GetLabels()[clusterv1.WatchLabel]
```